multinode/adapter_test.go

```diff
@@ -0,0 +1,203 @@
+package multinode
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	common "github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink-framework/multinode/config"
+)
+
+type testRPC struct {
+	latestBlock int64
+}
+
+type testHead struct {
+	blockNumber int64
+}
+
+func (t *testHead) BlockNumber() int64           { return t.blockNumber }
+func (t *testHead) BlockDifficulty() *big.Int    { return nil }
+func (t *testHead) GetTotalDifficulty() *big.Int { return nil }
+func (t *testHead) IsValid() bool                { return true }
+
+func LatestBlock(ctx context.Context, rpc *testRPC) (*testHead, error) {
+	rpc.latestBlock++
+	return &testHead{rpc.latestBlock}, nil
+}
+
+func ptr[T any](t T) *T {
+	return &t
+}
+
+func newTestClient(t *testing.T) *Adapter[testRPC, *testHead] {
+	requestTimeout := 5 * time.Second
+	lggr := logger.Test(t)
+	cfg := &config.MultiNodeConfig{
+		MultiNode: config.MultiNode{
+			Enabled:                      ptr(true),
+			PollFailureThreshold:         ptr(uint32(5)),
+			PollInterval:                 common.MustNewDuration(15 * time.Second),
+			SelectionMode:                ptr(NodeSelectionModePriorityLevel),
+			SyncThreshold:                ptr(uint32(10)),
+			LeaseDuration:                common.MustNewDuration(time.Minute),
+			NodeIsSyncingEnabled:         ptr(false),
+			NewHeadsPollInterval:         common.MustNewDuration(5 * time.Second),
+			FinalizedBlockPollInterval:   common.MustNewDuration(5 * time.Second),
+			EnforceRepeatableRead:        ptr(true),
+			DeathDeclarationDelay:        common.MustNewDuration(20 * time.Second),
+			NodeNoNewHeadsThreshold:      common.MustNewDuration(20 * time.Second),
+			NoNewFinalizedHeadsThreshold: common.MustNewDuration(20 * time.Second),
+			FinalityTagEnabled:           ptr(true),
+			FinalityDepth:                ptr(uint32(0)),
+			FinalizedBlockOffset:         ptr(uint32(50)),
+		},
+	}
+	c := NewAdapter[testRPC, *testHead](cfg, &testRPC{}, requestTimeout, lggr, LatestBlock, LatestBlock)
+	t.Cleanup(c.Close)
+	return c
+}
+
+func TestMultiNodeClient_LatestBlock(t *testing.T) {
```

Comment:
It woulbe be nice to add more coverage to verify that OnNewHead and OnNewFinalizedHead properly treats HealthCheckRequests and respects closure of requestCh